include/grpcpp/impl/codegen/call_op_set.h

```diff
@@ -310,7 +310,13 @@ class CallOpSendMessage {
 
  protected:
   void AddOp(grpc_op* ops, size_t* nops) {
-    if (msg_ == nullptr && !send_buf_.Valid()) return;
+    if (msg_ == nullptr && !send_buf_.Valid()) {
+      if (send_message_called_) {
+        gpr_log(GPR_ERROR, "API misuse observed: sending invalid message");
```

Comment:
I think this is a problem; I think we're not allowed to use gpr_log in impl/codegen . This caused me to experience a rollback on a recent PR that I haven't yet restored. You can just assert false; the stack trace will say the line number of the failure anyway.

Reply:
Asserted on !send_message_called_.